gamla/io_utils.py

```diff
@@ -166,6 +166,19 @@ async def run(*args, **kwargs):
 
 
 @functional_generic.curry
-async def get_async(timeout, url):
+async def get_async(timeout: float, url: Text):
     async with httpx.AsyncClient() as client:
         return await client.get(url, timeout=timeout)
+
+
+@functional_generic.curry
+async def post_json_async(timeout: float, url: Text, payload):
+    """Expects payload to be a json object, and the response to be json as well."""
```

Comment:
should this be a string ? or a Dict ? or dataclass as well ?
consider updating the comment and to add typing

also add return type (which I think is a Dict)

Reply:
it is whatever httpx considers valid json